lib/carrierwave/orm/activerecord.rb

```diff
@@ -93,11 +93,11 @@ def reload(*)
 
         # Reset cached mounter on record dup
         def initialize_dup(other)
-          old_mounters = @_mounters
+          old_mounters = @_mounters if instance_variable_defined?(:@_mounters)
           @_mounters = nil
           super
-          old_mounters.each do |column, mounter|
-            _mounter(:#{column}).cache(mounter.uploaders)
+          old_mounters&.each do |column, mounter|
+            _mounter(column).cache(mounter.uploaders)
```

Comment:
Weirdly enough, these two lines have the side effect of fixing the method being defined in the wrong context.

You don't want this method defined unless you have at least one mount field in the model, so it should be added by this method builder somehow. The problem is, the method gets redefined for every mount field, and these two lines apply to all fields and you only want them to run once.

I could have tried to make another module for this and `reload` that only gets included once, and then check whether it's there before including it again. But, that would add a module name and that inclusion check is a little difficult to do compatibly across the Ruby versions supported.

However, I noticed that if this method is defined and called repeatedly in the same inheritance chain, `@_mounters` is nil already for every later call, due to it having been assigned nil in the earlier call. When `old_mounters` is nil, `old_mounters&.each` doesn't do anything. This means that the later calls are safe to run because they don't do anything significant that can't be repeated.

This is the same trick that makes the `reload` method above safe to repeat, as it only assigns a nil to the same variable once per field.